### PR TITLE
Updates to business acumen competency and adding explicit note for ML engineer section

### PR DESCRIPTION
**Pull request by jnyang**
- Minor updates for clarity of the business acumen criteria
- Added notes in System Design and Tech Design to specify that they're only applicable for the ML engineer role

### CODE REVIEW

File: competencies/roles/DataScience.md [hunk 1]
```diff
@@ -27,24 +27,24 @@ with [Developers](./Developer.md)/[DataEngineers](./DataEngineer.md) but also wi
...
+            <li>Able to participate in team discussions regarding analysis and capture key requirements. Capable of translating business questions to concise technical asks.</li>
+            <li>Effective at soliciting feedback on the effectiveness of analyses.</li>
         </ul></td>
         <td><ul>
             <!--- Proficient  -->
-            <li>Active participation and outreach in explorating an ambiguous space, mapping to technical space, and communicating implications effectively.</li>
-            <li>Can participate in a business level objectives discussion and develop testable data hypotheses.</li>
-            <li>Systematically approach questions of effectiveness and identify adjacent opportunities or questions;
-                systematically check understanding of problem and relevance of solutions.</li>
+            <li>Active participation and outreach in explorating an ambiguous problem space, mapping to technical space, and  effectively communicating implications of analyses or technical methods.</li>
```
Comment (rsiu) [inline competencies/roles/DataScience.md:36]: Can you fix the typo here too please? `explorating`

File: competencies/roles/DataScience.md [hunk 2]
```diff
@@ -27,24 +27,24 @@ with [Developers](./Developer.md)/[DataEngineers](./DataEngineer.md) but also wi
...
+                systematically checks understanding of problem and relevance of solutions.</li>
         </ul></td>
         <td><ul>
             <!--- Authority -->
-            <li>Effective at understanding and communicating limitations of "knowing more" when a problem is no longer a "know more" situation</li>
-            <li>Effective thought partner for decision makers, improving their skills and thinking overall</li>
-            <li>Explorating an ambiguous space, mapping to technical space, and communicating implications effectively and impactfully. Aligned to business strategy and value add</li>
-            <li>Helps to organize business and data alignment processes that uncover needs</li>
-            <li>Systematically approach questions of effectiveness and identify adjacent opportunities or questions; develop ideas for improving overall data program in understanding of problem and relevance of solutions</li>
+            <li>Effective at understanding when a problem space has been sufficiently explorated and communicating the limitations of continuing to explorate</li>
```
Comment (rsiu) [inline competencies/roles/DataScience.md:43]: This competency grid is starting to make me question whether `explorate` is a real word or not 😆

File: competencies/roles/DataScience.md [hunk 3]
```diff
@@ -27,24 +27,24 @@ with [Developers](./Developer.md)/[DataEngineers](./DataEngineer.md) but also wi
...
+            <li>Effective thought partner for decisionmakers, improving their knowledge and critical thinking</li>
```
Comment (rsiu) [inline competencies/roles/DataScience.md:44]: decision makers

File: competencies/roles/DataScience.md [hunk 4]
```diff
@@ -27,24 +27,24 @@ with [Developers](./Developer.md)/[DataEngineers](./DataEngineer.md) but also wi
...
         </ul></td>
         <td><ul>
             <!--- Proficient  -->
-            <li>Active participation and outreach in explorating an ambiguous space, mapping to technical space, and communicating implications effectively.</li>
-            <li>Can participate in a business level objectives discussion and develop testable data hypotheses.</li>
-            <li>Systematically approach questions of effectiveness and identify adjacent opportunities or questions;
-                systematically check understanding of problem and relevance of solutions.</li>
+            <li>Active participation and outreach in explorating an ambiguous problem space, mapping to technical space, and  effectively communicating implications of analyses or technical methods.</li>
+            <li>Able to participate in business level objectives discussions and develop testable data hypotheses.</li>
+            <li>Systematically approaches analyses with questions of effectiveness and identify adjacent opportunities or questions;
```
Comment (rsiu) [inline competencies/roles/DataScience.md:38]:
```suggestion
            <li>Systematically approaches analyses with questions of effectiveness and identifies adjacent opportunities or questions;
```

File: competencies/roles/DataScience.md (hunk 2 above)
Comment (rsiu) [inline competencies/roles/DataScience.md:43]:
```suggestion
            <li>Effective at understanding when a problem space has been sufficiently explored and communicating the limitations of continuing to explore</li>
```

File: competencies/roles/DataScience.md (hunk 3 above)
Comment (rsiu) [inline competencies/roles/DataScience.md:44]:
```suggestion
            <li>Effective thought partner for decision makers, improving their knowledge and critical thinking</li>
```

File: competencies/roles/DataScience.md [hunk 5]
```diff
@@ -27,24 +27,24 @@ with [Developers](./Developer.md)/[DataEngineers](./DataEngineer.md) but also wi
...
-            <li>Effective thought partner for decision makers, improving their skills and thinking overall</li>
-            <li>Explorating an ambiguous space, mapping to technical space, and communicating implications effectively and impactfully. Aligned to business strategy and value add</li>
-            <li>Helps to organize business and data alignment processes that uncover needs</li>
-            <li>Systematically approach questions of effectiveness and identify adjacent opportunities or questions; develop ideas for improving overall data program in understanding of problem and relevance of solutions</li>
+            <li>Effective at understanding when a problem space has been sufficiently explorated and communicating the limitations of continuing to explorate</li>
+            <li>Effective thought partner for decisionmakers, improving their knowledge and critical thinking</li>
+            <li>Explorating an ambiguous problem space, mapping to technical space, and communicating implications effectively and impactfully. Aligned to business strategy and value add</li>
```
Comment (rsiu) [inline competencies/roles/DataScience.md:45]:
```suggestion
            <li>Exploring an ambiguous problem space, mapping to technical space, and communicating implications effectively and impactfully. Aligned to business strategy and value add</li>
```

File: competencies/roles/DataScience.md [hunk 6]
```diff
@@ -27,24 +27,24 @@ with [Developers](./Developer.md)/[DataEngineers](./DataEngineer.md) but also wi
...
+            <li>Helps to organize the business-and-data alignment processes and uncover areas for improvement</li>
+            <li>Systematically approaches questions of effectiveness and identify adjacent opportunities or questions; develops ideas for improving overall data program in understanding of problem and relevance of solutions</li>
```
Comment (rsiu) [inline competencies/roles/DataScience.md:47]:
```suggestion
            <li>Systematically approaches questions of effectiveness and identifies adjacent opportunities or questions; develops ideas for improving overall data program in understanding of problem and relevance of solutions</li>
```

File: competencies/roles/DataScience.md (hunk 1 above)
Comment (jnyang) [reply competencies/roles/DataScience.md:36]: Explorating is an actual word, but can change it to exploring if it's more clear

**Review by rsiu — APPROVED** [review]

**Pull request merged**